### ISSUE TITLE
Wrong behavior in 6th device screen while trying to remove a device

### ISSUE THREAD

**Issue by data-sync-user** [mozilla-mobile/mozilla-vpn-client#4792]
**Mozilla VPN affected versions:**

* 2.11.0 (2.202210270555)

**Platform tested:**

* Mac OS 10.15;
* Windows 10/11;

**Prerequisites:**

* Have the Mozilla VPN installed and opened;
* Have the current device, previously removed;
* Be logged in on 5 devices;

**Steps to reproduce:**

1. Log in with an account with subscription on the current device;
1. In the 6th device screen, remove a device;
1. Observe the behavior;

**Expected results:**

* Device is removed, user is redirected to the activation screen;

**Actual results:**

* User is redirected to the Get Started screen and the auth error rede banner is displayed.

**Notes:**

* Not reproducing if current device was not previously removed;
* Not reproducing on Linux;
* On Android user remains on 6th device screen ;
* Attaching logs from the client and from the app folder;

┆Issue is synchronized with this [Jira Bug](https://mozilla-hub.atlassian.net/browse/VPN-3144)
┆Reporter: Valentina Virlics

**Comment by data-sync-user** [comment]
➤ Beatriz Rizental Machado commented:

I believe this is a duplicate of [VPN-78].

There are currently two concurrent ways of receiving the message that a device has been removed: push message or task. They are running in parallel and sometimes one can superseed the other (see https://mozilla-hub.atlassian.net/browse/VPN-2730 ( https://mozilla-hub.atlassian.net/browse/VPN-2730|smart-link )). This race condition is expected and not necessarily a bug, until push messages are validated as a feature we will keep using both methods.

This is related to that because the “Authentication error” will be shown when the device is deleted through a task and will not be shown when it’s deleted through a push message.

In summary, https://mozilla-hub.atlassian.net/browse/VPN-78 ( https://mozilla-hub.atlassian.net/browse/VPN-78|smart-link ) is not fixed until push messages are the sole way of deleting a device and that will not happen until the feature is validated in prod. Either way we can close this one is favour of that one or vice versa. Does this make sense to you Valentina Virlics ?

**Comment by data-sync-user** [comment]
➤ Beatriz Rizental Machado commented:

After talking to Valentina Virlics on Slack and looking at the logs again I remove my comment about this being a duplicate. I looks a separate issue indeed. I can take a look at it in the next days.

**Comment by data-sync-user** [comment]
➤ Beatriz Rizental Machado commented:

Unassinging myself as I have no plans to work on this at the moment.

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

Re-verified this while using latest Mozilla VPN 2.21.0 (2.202402142134).

User is redirected to activation screen after being in 6th device screen, even if the device was previously removed.

Marking this as fixed.
